src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    const data = actions[i];
+
+    if (data.type === 'addProperties') {
+      Object.assign(state, data.extraData);
+    }
+
+    if (data.type === 'removeProperties') {
+      for (let j = 0; j < data.keysToRemove.length; j++) {
+        delete state[data.keysToRemove[j]];
+      }
+    }
+
+    if (data.type === 'clear') {
+      Object.keys(state).forEach((key) => {
+        delete state[key];
+      });
+    }
+    resultArray.push({ ...state });
```

Comment:
Pushing a copy of the object to the array should be done at the end of each loop cycle but outside of the `switch` block, after the current action is processed. [CODE STYLE]